src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -1,16 +1,879 @@
+        if self.green_loop.get_pressure()
+            <= Pressure::new::<psi>(A320Hydraulic::MIN_PRESS_PRESSURISED_LO_HYST)
+        {
+            self.is_green_pressurised = false;
+        } else if self.green_loop.get_pressure()
+            >= Pressure::new::<psi>(A320Hydraulic::MIN_PRESS_PRESSURISED_HI_HYST)
+        {
+            self.is_green_pressurised = true;
+        }
+
+        if self.blue_loop.get_pressure()
+            <= Pressure::new::<psi>(A320Hydraulic::MIN_PRESS_PRESSURISED_LO_HYST)
+        {
+            self.is_blue_pressurised = false;
+        } else if self.blue_loop.get_pressure()
+            >= Pressure::new::<psi>(A320Hydraulic::MIN_PRESS_PRESSURISED_HI_HYST)
+        {
+            self.is_blue_pressurised = true;
+        }
+
+        if self.yellow_loop.get_pressure()
+            <= Pressure::new::<psi>(A320Hydraulic::MIN_PRESS_PRESSURISED_LO_HYST)
+        {
+            self.is_yellow_pressurised = false;
+        } else if self.yellow_loop.get_pressure()
+            >= Pressure::new::<psi>(A320Hydraulic::MIN_PRESS_PRESSURISED_HI_HYST)
+        {
+            self.is_yellow_pressurised = true;
         }
     }
 
     pub fn is_blue_pressurised(&self) -> bool {
-        self.blue_pressurised
+        self.is_blue_pressurised
+    }
+
+    pub fn is_green_pressurised(&self) -> bool {
+        self.is_green_pressurised
+    }
+
+    pub fn is_yellow_pressurised(&self) -> bool {
+        self.is_yellow_pressurised
+    }
+
+    pub fn update(
+        &mut self,
+        ct: &UpdateContext,
+        engine1: &Engine,
+        engine2: &Engine,
+        overhead_panel: &A320HydraulicOverheadPanel,
+    ) {
+        let min_hyd_loop_timestep = Duration::from_millis(A320Hydraulic::HYDRAULIC_SIM_TIME_STEP); //Hyd Sim rate = 10 Hz
+
+        self.total_sim_time_elapsed += ct.delta;
+
+        //time to catch up in our simulation = new delta + time not updated last iteration
+        let time_to_catch = ct.delta + self.lag_time_accumulator;
+
+        //Number of time steps to do according to required time step
+        let number_of_steps_f64 = time_to_catch.as_secs_f64() / min_hyd_loop_timestep.as_secs_f64();
+
+        //updating rat stowed pos on all frames in case it's used for graphics
+        self.rat.update_stow_pos(&ct.delta);
+
+        if number_of_steps_f64 < 1.0 {
+            //Can't do a full time step
+            //we can either do an update with smaller step or wait next iteration
+
+            self.lag_time_accumulator =
+                Duration::from_secs_f64(number_of_steps_f64 * min_hyd_loop_timestep.as_secs_f64());
+        //Time lag is float part of num of steps * fixed time step to get a result in time
+        } else {
+            let num_of_update_loops = number_of_steps_f64.floor() as u32; //Int part is the actual number of loops to do
+                                                                          //Rest of floating part goes into accumulator
```

Comment:
Yup! Got lost with the fmt stuff screwing it up automatically :)  Comment before avoids this.

As for the comment believe me: if it won't help you it will probably help someone at some point.